socorro/external/es/crashstorage.py

```diff
@@ -208,9 +211,182 @@ def save_raw_and_processed(self, raw_crash, dumps, processed_crash,
+            'maximum_queue_size',
+            default=512,
+            doc='the maximum size of the internal queue'
+        )
+
+        #----------------------------------------------------------------------
+        def __init__(self, config, quit_check_callback=None):
+            super(ESBulkClassTemplate, self).__init__(
+                config,
+                quit_check_callback
+            )
+
+            self.task_queue = QueueWrapper(config.maximum_queue_size)
+            self.consuming_thread = Thread(
+                name="ConsumingThread",
+                target=self._consuming_thread_func
+            )
+
+            # overwrites original
+            self.transaction = config.transaction_executor_class(
+                config,
+                QueueContextSource(self.task_queue),
+                quit_check_callback
+            )
+            self.done = False
+            self.consuming_thread.start()
+
+        #----------------------------------------------------------------------
+        def _submit_crash_to_elasticsearch(self, queue, crash_document):
+            # Massage the crash such that the date_processed field is formatted
+            # in the fashion of our established mapping.
+            # First create a datetime object from the string in the crash
+            # report.
+            crash_date = datetimeutil.string_to_datetime(
+                crash_document['processed_crash']['date_processed']
+            )
+            # Then convert it back to a string with the expected formatting.
+            crash_date_with_t = datetimeutil.date_to_string(crash_date)
+            # Finally, re-insert that string back into the report for indexing.
+            crash_document['processed_crash']['date_processed'] = \
+                crash_date_with_t
+
+            # Obtain the index name.
+            es_index = self.get_index_for_crash(crash_date)
+            es_doctype = self.config.elasticsearch.elasticsearch_doctype
+            crash_id = crash_document['crash_id']
+
+            # Attempt to create the index; it's OK if it already exists.
+            if es_index not in self.indices_cache:
+                index_creator = IndexCreator(config=self.config)
+                index_creator.create_socorro_index(es_index)
+
+            action = {
+                '_index': es_index,
+                '_type': es_doctype,
+                '_id': crash_id,
+                '_source': crash_document,
+            }
+            queue.put(action)
+
+        #----------------------------------------------------------------------
+        def _consumer_iter(self):
+            try:
+                while True:
+                    crash_document = self.task_queue.get()
+                    if crash_document is None:
+                        self.done = True
+                        break
+                    try:
+                        yield crash_document  # execute the task
+                    except Exception:
```

Comment:
Should this not capture more specific exceptions? I suppose you want to catch problems coming from the ``elasticsearch`` library, so that would be ``elasticsearch.exceptions.ElasticsearchException``.